scripts/dts/extract/globals.py

```diff
@@ -331,6 +331,54 @@ def build_cell_array(prop_array):
 
     return ret_array
 
+def child_to_parent_unmap(cell_parent, gpio_index):
+    # Aim of this function is to return a gpio-controller from the
+    # cell_parent provided (which has been identified as a 'nexus node',
+    # ie: has a 'gpio-map' property) thanks to the gpio_index provided.
+    # Note: Nexus nodes and gpio-map property are described in devide tree
+    # specification, v0.3, chapter 'Nexus nodes and Specifier Mapping'.
+
+    # First, retrieve gpio-map as a list
+    raw_map = reduced[cell_parent]['props']['gpio-map']
+
+    # We now need to build the map as an array of array,
+    # which will allow to use provided index
+
+    # map raws are encoded as follows:
+    # [child specifier][gpio controller phandle][parent specifier]
+    # child specifier field length is connector property #gpio-cells
+    # parent specifier field length is parent property #gpio-cells
+
+    child_specifier_size = reduced[cell_parent]['props']['#gpio-cells']
+    # Assumption 1: We assume parent #gpio-cells is constant across
+    # the map, so we take the value of the first occurrence and apply
+    # to the whole map.
+    parent = phandles[child_specifier_size+1]
```

Comment:
I don't get this part. How is `child_specifier_size` ("The number of 32-bit cells required to specify this component", according to the DT spec) related to phandles?

Reply:
gpio-map is encoded in the following way:
```
        gpio-map = <0 0 &soc_gpio1 1 0>,                   
                   <1 0 &soc_gpio2 4 0>,
                   <2 0 &soc_gpio1 3 0>,
                   <3 0 &soc_gpio2 2 0>;
```
Taking the first line:
0 0: Child specifier (pin number, pin flag)
&soc_gpio2: GPIO controller phandle
1 0: Parent specifier (pin number, pin flag)

At this point we're looking for GPIO controller phandle which is the element number 'child_specifier_size+1' in the line.

About gpio-map pin flags (does not relates directly to your question, but this might be required for the full comprehension of the change):
Assuming optional params `gpio-map-mask` and `gpio-map-pass-thru` are omitted, child and parent specifier pin flags are not taken into account and pin flag used is the one provided in calling node (typically used in shield overlay file), which is the behavior we expect, at least in current shield use case. This is "Assumption 2".

Reply:
AFAICS, this is looking at `#gpio-cells` though. It's saying "the number of GPIO cells + 1 is a phandle". Seems weird to me.

Sorry if I'm missing something.

Reply:
So:
0 0: Child specifier (pin number, pin flag) -> length provided by current node's #gpio-cell
&soc_gpio2: GPIO controller phandle
1 0: Parent specifier (pin number, pin flag)  -> length provided by GPIO controller's #gpio-cell

Reply:
You're never indexing into `gpio-map` though. You're taking `#gpio-cells + 1` and directly treating that as a phandle.

Reply:
Indeed, Txs!